lib/spack/llnl/util/tty/log.py

```diff
@@ -783,12 +787,16 @@ def _writer_daemon(stdin_multiprocess_fd, read_multiprocess_fd, write_fd, echo,
                             # find control characters and strip them.
                             clean_line, num_controls = control.subn('', line)
 
+                            # apply any user-provided output filter
+                            if filter_fn:
+                                filtered = filter_fn(clean_line)
+
                             # Echo to stdout if requested or forced.
                             if echo or force_echo:
-                                sys.stdout.write(clean_line)
+                                sys.stdout.write(filtered)
 
                             # Stripped output to log file.
-                            log_file.write(_strip(clean_line))
+                            log_file.write(_strip(filtered))
```

Comment:
Like we discussed earlier today, this should maybe me the *unfiltered* output that goes to the file, to match the PR description.